src/oemof_tabular_plugins/general/post_processing/post_processing.py

```diff
@@ -383,13 +438,93 @@ def create_storage_capacities_table(all_scalars, results):
     return storage_capacities_df
 
 
+def calculate_total_land_requirement(results, capacities_df, storage_capacities_df):
+    # ToDo: this parameter should only be displayed in the results if the parameters have been defined in the
+    #  CSV input files
+    """
+    Calculates the total land requirement needed for the energy system (existing, planned (fixed) and optimized capacities).
+    :param results: oemof model results
+    :param capacities_df: capacities dataframe
+    :param storage_capacities_df: storage capacities dataframe
+    :return: total land requirement value
+    """
+    # initiate total land requirement value
+    total_land_requirement = 0
+    # set boolean for finding land requirement parameter in any of the csv inputs
+    land_requirement_found = False
+    # convert capacities_df['Component'] column to a list
+    component_names = capacities_df["Component"].tolist()
+    # convert storage_capacities_df['Component'] column to a list
+    storage_component_names = storage_capacities_df["Component"].tolist()
+    # loop through the results dict
+    for entry_key, entry_value in results.items():
+        component_name = entry_key[0]
+        # check if the oemof object tuple has the 'output_parameters' attribute
+        if hasattr(component_name, "output_parameters"):
+            # store the 'output_parameters' dict as output_param_dict
+            output_param_dict = component_name.output_parameters
+            # retrieve the 'land_requirement' value if it exists
+            # NOTE: this means that the user must define the land requirement as 'land_requirement' otherwise
+            # it won't get considered in the total land requirement value
+            land_requirement = output_param_dict.get("custom_attributes", {}).get(
+                "land_requirement"
+            )
+            if land_requirement is not None and str(component_name) in component_names:
+                # set to True because a land requirement parameter has been found
+                land_requirement_found = True
+                # retrieve the total capacity from capacities_df and calculate total land requirement
+                total_capacity = capacities_df.loc[
+                    capacities_df["Component"] == str(component_name), "Total Capacity"
+                ].iloc[0]
+                total_component_land_requirement = land_requirement * total_capacity
+                total_land_requirement += total_component_land_requirement
```

Comment:
```suggestion
                component_land_requirement = land_requirement * total_capacity
                total_land_requirement += component_land_requirement
```

I find it easier to see it is two different variables

Reply:
agreed, done